File: euler6.hs
{-Sum square difference
Problem 6

The sum of the squares of the first ten natural numbers is,
12 + 22 + ... + 102 = 385

The square of the sum of the first ten natural numbers is,
(1 + 2 + ... + 10)2 = 552 = 3025

Hence the difference between the sum of the squares of the first ten natural numbers and the square of the sum is 3025 − 385 = 2640.

Find the difference between the sum of the squares of the first one hundred natural numbers and the square of the sum.
-}

sumsquared = sum (map ((^2)) [1,2..100])
squaredsum = (sum [1,2..100]) ^ 2

result = squaredsum - sumsquared

main = do
  print result

File: euler6.clj
(defn square
  [x]
  (* x x))

(defn sum-squares
  []
  (reduce + (map square (range 1 101))))

(defn squared-sum
  []
  (square (reduce + (range 1 101))))

(defn result
  []
  (- (squared-sum) (sum-squares)))
